diff --git a/mingw-w64-libass/PKGBUILD b/mingw-w64-libass/PKGBUILD
index 49ed91154..9f4fc253d 100644
--- a/mingw-w64-libass/PKGBUILD
+++ b/mingw-w64-libass/PKGBUILD
@@ -21,6 +21,20 @@ options=('strip' 'staticlibs')
 source=(https://github.com/libass/${_realname}/releases/download/${pkgver}/${_realname}-${pkgver}.tar.xz)
 sha256sums=('1cdd39c9d007b06e737e7738004d7f38cf9b1e92843f37307b24e7ff63ab8e53')
 
+prepare() {
+    cd "${srcdir}"
+    if [[ $MINGW_ARCH == "mingw64" ]]
+    then
+        curl "https://github.com/lifegpc/static-mingw-w64-fribidi/releases/download/1.0.10-2/mingw-w64-x86_64-fribidi-1.0.10-2-any.pkg.tar.zst" -o fribidi.pkg.tar.zst -L
+        pacman -U fribidi.pkg.tar.zst --overwrite '*' --noconfirm
+    fi
+    if [[ $MINGW_ARCH == "mingw32" ]]
+    then
+        curl "https://github.com/lifegpc/static-mingw-w64-fribidi/releases/download/1.0.10-2/mingw-w64-i686-fribidi-1.0.10-2-any.pkg.tar.zst" -o fribidi.pkg.tar.zst -L
+        pacman -U fribidi.pkg.tar.zst --overwrite '*' --noconfirm
+    fi
+}
+
 build() {
   [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
   mkdir -p "${srcdir}/build-${MINGW_CHOST}"
@@ -30,10 +44,11 @@ build() {
     --build=${MINGW_CHOST} \
     --host=${MINGW_CHOST} \
     --target=${MINGW_CHOST} \
-    --enable-shared \
+    --enable-shared=no \
     --enable-static \
     --enable-fontconfig \
-    --enable-asm
+    --enable-asm \
+    CFLAGS="-DFRIBIDI_LIB_STATIC"
   make
 }
 
